app/updatemanager.go

```diff
@@ -373,7 +398,7 @@ func (c *ControlMapPool) Get(ID string) (active []*UpdateControlMap, expired []*
 		c.Pool,
 		// Collector
 		func(u *UpdateControlMap) {
-			if u.expired {
+			if u.Expired() {
```

Comment:
`Expirations` are now guarded by a separate `mutex`, which should resolve the issue we've been discussing